src/verify.c

```diff
@@ -108,11 +121,45 @@ static void print_help(const char *name)
 	fprintf(stderr, "\n");
 }
 
+static bool compile_whitelist()
+{
+	int errcode;
+	char *error_buffer = NULL;
+	char *full_regex = NULL;
+	bool success = false;
+
+	/* Enforce matching the entire path. */
+	string_or_die(&full_regex, "^(%s)$", cmdline_option_picky_whitelist);
+
+	assert(!picky_whitelist);
```

Comment:
Why fail here if picky_whitelist is set? You only call compile_whitelist at one point so I don't see how it could be set.